core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
@guozhangwang I hadn't pushed the changes last night, so the code is the same as it was. I have fixed the comment though. I think it should be ok in these scenarios:
* Old consumer timeout/exception: `consumerConnector.shutdown()` method is called in finally block and in shutdown hook, but that works fine
* Old consumer Ctrl-C - `consumerConnector.shutdown()` method is called in shutdown hook
* New consumer timeout/exception: `consumer.close()` is called in finally block, the `wakeup()` in shutdown hook has no effect
* New consumer Ctrl-C: Shutdown hook calls `wakeup()` that terminates the receive thread. `close()` cannot be invoked safely from shutdown hook without actually waiting for receive thread to finish. Hence `consumer.close()` is not invoked.